test/unit-test/ota_utest.c

```diff
@@ -323,6 +347,29 @@ static OtaMqttStatus_t stubMqttPublish( const char * const unused_1,
     return OtaMqttSuccess;
 }
 
+OtaErr_t mockControlInterfaceRequestJobAlwaysFail( OtaAgentContext_t * unused )
+{
+    ( void ) unused;
+
+    return OtaErrRequestJobFailed;
+}
+
+
+OtaErr_t mockDataInterfaceInitFileTransferAlwaysFail( OtaAgentContext_t * unused )
+{
+    ( void ) unused;
+
+    return OtaErrInitFileTransferFailed;
+}
+
+
+OtaErr_t mockMqttInitFileTransferAlwaysSucceed( OtaAgentContext_t * unused )
```

Comment:
Can be repurposed for HTTP by renaming it to `mockDataInterfaceInitFileTransfer`

Reply:
done